File: src/IsbInterop/ComponentTokens/Wrappers/ComponentToken.vb
Imports IsbInterop.Base.Wrappers

Namespace ComponentTokens.Wrappers

  ''' <summary>
  ''' Обертка над IComponentToken.
  ''' </summary>
  Friend Class ComponentToken
    Inherits EdmsObject(Of IComponentTokenInfo)
    Implements IComponentToken

    ''' <summary>
    ''' Информация об объекте.
    ''' </summary>
    Public Overrides ReadOnly Property Info As IComponentTokenInfo
      Get
        Dim rcwIComponentTokenInfo = GetRcwObjectInfo()
        Return New ComponentTokenInfo(rcwIComponentTokenInfo, Scope)
      End Get
    End Property

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIComponentToken">СOM-объект IComponentToken.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIComponentToken As Object, scope As IScope)
      MyBase.New(rcwIComponentToken, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop.CSharp/ComponentTokens/Wrappers/ComponentToken.cs
using IsbInterop.Base.Wrappers;

namespace IsbInterop.ComponentTokens.Wrappers
{
  /// <summary>
  /// Обертка над IComponentToken.
  /// </summary>
  internal class ComponentToken : EdmsObject<IComponentTokenInfo>, IComponentToken
  {
    /// <summary>
    /// Получить IComponentTokenInfo.
    /// </summary>
    /// <returns>IComponentTokenInfo.</returns>
    public override IComponentTokenInfo GetInfo()
    {
      var rcwIComponentTokenInfo = this.GetRcwObjectInfo();
      return new ComponentTokenInfo(rcwIComponentTokenInfo, this.Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIComponentToken">СOM-объект IComponentToken.</param>
    /// <param name="scope">Область видимости.</param>
    public ComponentToken(object rcwIComponentToken, IScope scope) : base(rcwIComponentToken, scope) { }
  }
}
